solutions/moderation.py

```diff
@@ -0,0 +1,29 @@
+import logging
+import json
+# from lib.variables import get_environment_variable
+from lib.openaiutils import openai_get_authenticated_client, get_openai_models
+
+logger = logging.getLogger(__name__)
+
+def moderation(client, prompt):
+    moderation = client.moderations.create(input=prompt)
```

Comment:
Cool. I missed that client has the moderations(), so I've used in my solution just a POST request:

```python
import logging
import json
from lib.variables import get_environment_variable
from lib.openaiutils import openai_get_authenticated_client, get_openai_models
from lib.curlutils import send_request, HTTPMethod
from lib.variables import get_environment_variable, AIDEVS_OPENAI_API_KEY

logger = logging.getLogger(__name__)

MODERATION_MODEL = 'text-moderation-latest'


API_ENDPOINT = 'https://api.openai.com/v1/moderations'


def _string_to_moderation_input(input_string):
    json_data = {
        "input": input_string
    }
    return json_data

def _is_this_permitted(input_json):
    openai_api_key = get_environment_variable(AIDEVS_OPENAI_API_KEY)
    output_http = send_request(API_ENDPOINT, HTTPMethod.POST, input_json, openai_api_key)
    return output_http

def get_answer(task_details_response: dict):
    client = openai_get_authenticated_client()
    logger.info(f"Solving moderate task for input: {dict}")

    answer = []

    for question in task_details_response['input']:
        input_json = _string_to_moderation_input(question)
        valid = _is_this_permitted(input_json)
        json_data = json.loads(valid.text)
        get_answer = json_data['results'][0]['flagged']
        if get_answer:
            answer.append(1)
        else:
            answer.append(0)
    return answer
```